test/performance/throughput/ThroughputPublisher.cpp

```diff
@@ -70,11 +70,12 @@ void ThroughputPublisher::DataPubListener::onPublicationMatched(
         --throughput_publisher_.data_discovery_count_;
     }
 
+    lock.unlock();
+
     if (throughput_publisher_.data_discovery_count_ == static_cast<int>(throughput_publisher_.subscribers_))
     {
         throughput_publisher_.data_discovery_cv_.notify_one();
     }
```

Comment:
I'm not sure whether unlocking and the checking the count is a good idea. We could do something like:

```suggestion
    if (throughput_publisher_.data_discovery_count_ == static_cast<int>(throughput_publisher_.subscribers_))
    {
        lock.unlock();
        throughput_publisher_.data_discovery_cv_.notify_one();
        return;
    }
    lock.unlock();
```

Reply:
:+1: